### PR TITLE
Added Boot Sequence feature.

### PR DESCRIPTION
**Pull request by bond95**
Bug: When trying to run VM with no disks, but with one NIC, it throw error, because can't find bootable device.
Fixes: https://github.com/oVirt/ovirt-web-ui/issues/565

@jniederm @mareklibra @leistnerova What do you think about design?
image
image

---
This change is [Reviewable](https://reviewable.io/reviews/ovirt/ovirt-web-ui/575)

### CODE REVIEW

**Comment by jniederm** [comment]
It would be nice to have the numbering unified. Personally I prefer the "first device", "second device" labels option .

image

File: src/ovirtapi.js [hunk 1]
```diff
@@ -267,8 +269,13 @@ OvirtApi = {
         id: vm.cluster.id,
       } : undefined,
 
-      os: vm.os && vm.os.type ? {
-        type: vm.os.type,
+      os: vm.os && (vm.os.type || vm.os.bootDevices) ? {
+        type: vm.os.type || undefined,
+        boot: {
+          devices: {
+            device: vm.os.bootDevices.filter((item) => item !== null),
```
Comment (mareklibra) [inline src/ovirtapi.js]: weird case, but might happen: `vm.os.bootDevices === undefined`

**Comment by mareklibra** [comment]
This needs rebase and addressing the comment above

File: src/ovirtapi.js (hunk 1 above)
Comment (bond95) [reply src/ovirtapi.js]: Actually no, because it get data from VmDialog, and VmDialog in any case will return array.

**Comment by bond95** [comment]
@jniederm @mareklibra Done.
image

**Pull request merged**